ui/lib/apps/InstanceProfiling/pages/Detail.tsx

```diff
@@ -37,6 +69,100 @@ function isFinished(data) {
   return groupState === 2 || groupState === 3
 }
 
+async function getActionToken(
+  id: string,
+  apiType: string
+): Promise<string | undefined> {
+  const res = await client.getInstance().getActionToken(id, apiType)
+  const token = res.data
+  if (!token) {
+    return
+  }
+  return token
+}
+
+function ViewResultButton({ rec, t }) {
+  const isProtobuf: boolean =
+    rec.profile_output_type === ProfilingOutputTypeProtobuf
+  let token: string | undefined
+
+  const handleViewResultMenuClick = usePersistFn(async (e: MenuInfo) => {
+    switch (e.key) {
+      case 'download':
+        token = await getActionToken(rec.id, 'single_download')
+
+        window.location.href = `${client.getBasePath()}/profiling/single/download?token=${token}`
+        break
+      default:
+        token = await getActionToken(rec.id, 'single_view')
+
+        let profileURL = `${client.getBasePath()}/profiling/single/view?token=${token}&output_type=graph`
+        window.open(`${profileURL}`, '_blank')
+    }
+  })
+
+  const handleViewResultBtnClick = usePersistFn(async () => {
+    token = await getActionToken(rec.id, 'single_view')
+
+    let profileURL = `${client.getBasePath()}/profiling/single/view?token=${token}`
+    if (isProtobuf) {
+      const titleOnTab = rec.target.kind + '_' + rec.target.display_name
+      profileURL = `/dashboard/speedscope#profileURL=${encodeURIComponent(
+        profileURL + `&output_type=flamegraph`
+      )}&title=${titleOnTab}`
+    }
+
+    window.open(`${profileURL}`, '_blank')
+  })
+
+  const menu = () => {
+    return (
+      <Menu onClick={handleViewResultMenuClick}>
+        <Menu.Item key={ViewOutputTypeOptions.Graph}>
+          {t('instance_profiling.detail.table.columns.selection.view')}{' '}
+          {t(ViewOutputTypeOptions.Graph)}
+        </Menu.Item>
```

Comment:
This is very complicated and is not approachable when you have different source types. The previous design `view_output_options` (which is an array of options to be displayed) is more reasonable, since it is a design works for now and future.